### PR TITLE
Feature/modules

### PR DESCRIPTION
**Pull request by ranjan-stha**
Added modules and handlers for Summarization, NGrams, TopicModeling. Also, added the mockserver.

### CODE REVIEW

File: handlers/ecs/geolocations/app.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    Geolocation class to extract geolocations from the excerpts
+    """
+    def __init__(self):
+        self.entries_url = os.environ.get("ENTRIES_URL") or None
+        self.client_id = os.environ.get("CLIENT_ID") or None
+        self.callback_url = os.environ.get("CALLBACK_URL") or None
+        self.geolocation_id = os.environ.get("GEOLOCATION_ID",) or None
+        self.aws_region = os.environ.get("AWS_REGION", "us-east-1")
+        self.signed_url_expiry_secs = os.environ.get("SIGNED_URL_EXPIRY_SECS", 86400) # 1 day
+        self.bucket_name = os.environ.get("S3_BUCKET_NAME", None)
```
Comment (bewakes) [inline handlers/ecs/geolocations/app.py]: What is the purpose of using `.get(...) or None` vs `.get(..SECS, 86400)`? Why not `or` in both or default value in the function call itself?

File: handlers/ecs/geolocations/app.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+                "excerpt": str
+            }
+        ]
+        """
+        if self.entries_url:
+            logging.info("The request url is %s", self.entries_url)
+            try:
+                response = requests.get(self.entries_url, timeout=30)
+                if response.status_code == 200:
+                    entries_data = json.loads(response.text)
```
Comment (bewakes) [inline handlers/ecs/geolocations/app.py]: We can probably use `response.json()` as well.

File: handlers/ecs/geolocations/app.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+            geolocation_results = geolocation.get_geolocation_api(
+                raw_data=[in_entries_dict["excerpt"] for in_entries_dict in self.entries],
+                geonames_username=self.geoname_api_user
+            )
+            processed_results = [{
+                "entry_id": x["entry_id"],
+                "entities": y["entities"]
+                } for x, y in zip(self.entries, geolocation_results)
+            ]
+            date_today = str(datetime.now().date())
```
Comment (bewakes) [inline handlers/ecs/geolocations/app.py]: Maybe use ISO format like: `date.isoformat()`? this will probably give more consistent result.

**Review by bewakes — CHANGES_REQUESTED** [review]

File: handlers/ecs/geolocations/app.py (hunk 3 above)
Comment (aembryonic) [reply handlers/ecs/geolocations/app.py]: i think that str(DATE), or DATE.isoformat() will give the same result

File: handlers/ecs/geolocations/app.py (hunk 2 above)
Comment (ranjan-stha) [reply handlers/ecs/geolocations/app.py]: done.

File: handlers/ecs/geolocations/app.py (hunk 3 above)
Comment (ranjan-stha) [reply handlers/ecs/geolocations/app.py]: done

File: handlers/ecs/geolocations/app.py (hunk 1 above)
Comment (ranjan-stha) [reply handlers/ecs/geolocations/app.py]: done

**Comment by aembryonic** [comment]
well done

**Review by aembryonic — COMMENTED** [review]
well done

**Pull request merged**